crates/pathfinder/src/sequencer.rs

```diff
@@ -65,6 +65,44 @@ async fn parse_raw(resp: reqwest::Response) -> Result<reqwest::Response, Sequenc
     Ok(resp)
 }
 
+/// Wrapper function to allow retrying sequencer queries in an exponential manner.
+///
+/// Initial backoff time is 2 seconds. Retrying stops after approximately 4 minutes in total.
+async fn retry<T, Fut, FutureFactory>(future_factory: FutureFactory) -> Result<T, SequencerError>
+where
+    Fut: Future<Output = Result<T, SequencerError>>,
+    FutureFactory: FnMut() -> Fut,
+{
+    use crate::retry::Retry;
+    use reqwest::StatusCode;
+    use std::num::{NonZeroU64, NonZeroUsize};
+
+    Retry::exponential(future_factory, NonZeroU64::new(2).unwrap())
+        // Max number of retries of 7 gives a total accumulated timeout of 4 minutes and 15 seconds (2^8-1)
+        .max_num_retries(NonZeroUsize::new(7).unwrap())
+        .when(|e| match e {
+            SequencerError::TransportError(te) if te.is_timeout() => {
+                tracing::debug!("Retrying due to timeout");
+                true
+            }
+            SequencerError::TransportError(te) => match te.status() {
+                Some(
+                    status @ (StatusCode::TOO_MANY_REQUESTS
+                    | StatusCode::REQUEST_TIMEOUT
```

Comment:
Are the `is_timeout` and `REQUEST_TIMEOUT` matching different events?

Reply:
Let me check that :eyes: